webapp/src/main/java/com/box/l10n/mojito/service/translationkit/TranslationKitService.java

```diff
@@ -102,6 +103,13 @@ public TranslationKitAsXliff generateTranslationKitAsXLIFF(Long dropId, Long tmI
         logger.debug("Get translation kit for in tmId: {} and locale: {}", tmId, localeId);
         TranslationKit translationKit = addTranslationKit(dropId, localeId, type);
 
+        Long repositoryId = translationKit.getDrop().getRepository().getId();
+        StatusFilter statusFilter = TranslationKit.Type.TRANSLATION.equals(type) ? StatusFilter.FOR_TRANSLATION : StatusFilter.REVIEW_NEEDED;
+
+        if (isEmpty(repositoryId, localeId, statusFilter, useInheritance)) {
+            return null;
```

Comment:
I don't think it is a good pattern to return "null" to convey the notion of an empty translation kit. I would keep creating the object as before but have an attribute on TranslationKitAsXliff that says it is empty which can be used to export or not the file